hubspot-style/src/main/java/com/hubspot/immutables/utils/WireSafeEnum.java

```diff
@@ -113,6 +116,21 @@ public String asString() {
     return enumValue;
   }
 
+  @Nonnull
+  public T asEnumOrThrow() {
```

Comment:
I wonder if we want an overload that accepts a `Supplier<Throwable>`?

Reply:
In that case, couldn't they just call getOrThrow directly on the optional?